ThreeBotPackages/tft_statistics/services/updatestats.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+import os
+import sys
+
+from jumpscale.loader import j
+from jumpscale.tools.servicemanager.servicemanager import BackgroundService
+
+current_full_path = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(current_full_path + "/../../")
+
+from tft_statistics.sals.stats_sals import (
+    update_foundation_wallets_data,
+    update_stats,
+    update_total_tft,
+    update_total_unlocked_tft,
+)
+
+
+class UpdateTokensStats(BackgroundService):
+    def __init__(self, interval=60 * 60, *args, **kwargs):
+        """Update tokens stats every 60 minutes
+        """
+        super().__init__(interval, *args, **kwargs)
+        self.schedule_on_start = True
+
+    def job(self):
+        try:
+            # update foundation wallets
+            j.logger.info(f"Updating foundation wallets")
+            foundation_wallets = update_foundation_wallets_data()
+            j.logger.debug(f"Foundation wallets updated successfully {foundation_wallets}")
+
+            # update total tft & tfta total tokens
+            j.logger.info("Updating total TFT")
+            total_tft = update_total_tft(tokencode="TFT")
```

Comment:
Can't we derive this from the the detailed tft statistics which are calculated later?

Reply:
done